src/main/groovy/io/github/cosmicsilence/scalafix/ScalafixPlugin.groovy

```diff
@@ -81,33 +90,16 @@ class ScalafixPlugin implements Plugin<Project> {
             scalafixTask.compileOptions.set(project.provider({ sourceSet.compilerOptions }))
             scalafixTask.semanticdbConfigured = extension.semanticdb.autoConfigure.get()
 
+            sourceSet.compileTask.dependsOn configureSemanticdbCompilerTask
```

Comment:
could we have a comment here explaining why this is done inside `register` and not outside?